### DISCUSSION TITLE
Tag naming convensions

### DISCUSSION THREAD

**Discussion by gaohoward** [artemiscloud/activemq-artemis-self-provisioning-plugin#209] [General]
Currently there are two popular naming conventions for tag names. One is <major-version>-<minor-version>-<micro/patch-version> the other is prefix the previous one with a `v`.
For example tag for version 0.1.0 could be `0.1.0` or `v0.1.0`.
Which one should be used? I'd suggest we use the first one (`0.1.0`) to be consistent with other artemiscloud projects, such as the operator https://github.com/artemiscloud/activemq-artemis-operator/tags

**Comment by lavocatt** [comment]
I think this is a good idea. We would have to make [the release](https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/blob/main/.github/workflows/release.yml) support it.

**Comment by lavocatt** [comment]
Also something to note is that for upgrading to the dynamic plugin version 1.3.0 we will have to drop the `v` in the version name in the `package.json` file. Maybe that would be a good opportunity to do the switch to ` --<micro/patch-version>` ?

**Reply by Msarawan** [reply]
yeah, it's good.

**Comment by Msarawan** [comment]
Yeah, as Howard suggested i am agree with just keeping the tag as `0.1.0` to be consistent with other artemiscloud projects.

**Comment by lavocatt** [comment]
About the timeline, should we do this for the first version ? Because that would require a new PR (to update the github action) Or do we postpone this for the next release?

**Comment by RoddieKieley** [comment]
Aligning with the operator and being consistent will be important from an automation point of view, so as indicated above it's a matter of when. As the existing tag built successfully but experienced a [push failure ](https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/actions/runs/9675408545/job/26692869842#step:8:15)when attempting to push to the container registry maybe the thing to do is to update from v0.1.0 -> 0.1.0 now and then ensure that 0.1.0 gets pushed correctly to the registry for general consumption.

**Reply by lavocatt** [reply]
Opened a PR https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/pull/213

**Comment by lavocatt** [comment]
This is resolved. Thanks @howardgao for proposing this change.
